docs/06_how-to-guides/40_multi-index/how-to-insert-data-into-a-multi-index-table.md

```diff
@@ -1,22 +1,45 @@
 ---
-content_title: How to insert data into a multi index table
+content_title: How to insert data into a multi-index table
 ---
 
-## Preconditions
-- It is assumed you already have a multi index table instance defined along with its mandatory primary index, otherwise take a look at the section [How to instantiate a multi index table](./how-to-instantiate-a-multi-index-table.md).
+## Overview
 
-To insert data into a multi index table follow the following steps
+This guide provides instructions to insert data into a multi-index table.
+
+## Reference
+
+See the following code reference:
+
+* The [`multi-index`](../../classeosio_1_1multi__index) class.
+* The [`multi-index::find(...)`](../../group__multiindex#function-find) method.
+* The [`multi-index::emplace(...)`](../../group__multiindex/#function-emplace) method.
+
+## Before you begin
+
+Make sure you have the following prerequisites in place:
+
+* An EOSIO development environment, for details consult the [Get Started Guide](https://developers.eos.io/welcome/latest/getting-started-guide/index),
+* A multi-index `testab` table instance which stores `user` objects indexed by the primary key which is of type `eosio::name`. Consult the section [How to instantiate a multi-index table](./how-to-instantiate-a-multi-index-table.md) to learn how to set it up.
+
+## Procedure
+
+Complete the following steps to insert an user object in the `testtab` multi-index table:
+
+### 1. Verify If The User Already Exists
+
+Make use of the multi-index table iterator to find out if the user object already exists. The targeted user is searched based on its account name.
```

Comment:
Make use of the multi-index table iterator to find out if the user object already exists. The targeted user is searched based on its account name.

=>

Use the multi-index table iterator to find out if the user object already exists. The targeted user is searched based on its account name.